src/utils/format_json/open_telemetry.c

```diff
@@ -0,0 +1,285 @@
+/**
+ * collectd - src/utils_format_json.c
+ * Copyright (C) 2023       Florian octo Forster
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a
+ * copy of this software and associated documentation files (the "Software"),
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom the
+ * Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ *
+ * Authors:
+ *   Florian octo Forster <octo at collectd.org>
+ **/
+
+#include "collectd.h"
+
+#include "utils/format_json/format_json.h"
+
+#include <yajl/yajl_gen.h>
+
+#define CHECK(f)                                                               \
+  do {                                                                         \
+    int status = (f);                                                          \
+    if (status != 0) {                                                         \
+      if (strncmp(#f, "yajl_gen_", strlen("yajl_gen_")) == 0) {                \
+        ERROR("format_json: %s failed with status %d", #f, status);            \
+      }                                                                        \
```

Comment:
This whole thing is quite a lot of extra code for every CHECK line.  At least stop replicating this part by using something like:
```
static void log_yajl_gen_error(const char *fname) {
  if (strncmp(fname, "yajl_gen_", strlen("yajl_gen_")) == 0) {
    ERROR("format_json: %s failed with status %d", fname, status);
  }
}
```
and:
```suggestion
      log_yajl_gen_error(#f); \
```

PS. If the return value is relevant only for whether it's non-zero, another option would have been just to return NULL for success, and the failing function name for error.  Then print only in top level function that failing function name...

Reply:
Great suggestion, thank you! Done.

Reply:
Changes look good!